### DISCUSSION TITLE
posts/m1qemu

### DISCUSSION THREAD

**Discussion by giscus** [whexy/whexy-blog-comments#8] [General]
# posts/m1qemu

a student obsessed with the computing world

https://www.whexy.com/posts/m1qemu/

**Comment by jonathancross** [comment]
Nice!  Some questions:

* How does this compare with just using [UTM](https://mac.getutm.app/) ?
* Is nested virtualization fully supported (eg docker / KVM running on debian?)
* What is the performance hit?

**Reply by whexy** [reply]
Hi. Thank you for your reading.

> How does this compare with UTM?

QEMU is an emulation framework, and UTM is a virtual machine solution. I believe UTM uses QEMU as a backend to support x86 virtualization on M1, and they are definitely doing some other things to enhance the user experience. Like a friendly GUI or some tools for bridging host hosts and VMs. UTM is a better choice if you just want to run a virtual machine without debugging the kernel or tracking instructions.

> Is nested virtualization fully supported?

Docker works just fine on Linux VMs since it doesn't need hypervisor support. But KVM requires some hardware hypervisor features which QEMU cannot emulate (currently).
I also tried the crazy idea of running QEMU inside QEMU. And that works 😂.

> What is the performance hit?

Not quite good as expected, given the powerful performance of the M1 chip. Using an arm-based distribution of Linux is just fine. Running an intel-based distribution is terrible.

**Reply by jonathancross** [reply]
Thanks!

> But KVM requires some hardware hypervisor features which QEMU cannot emulate (currently).

That is disappointing.

Maybe you are interested in this: [nrjdalal/QEMU-Silicon-Mac-Virtualization.md](https://gist.github.com/nrjdalal/e70249bb5d2e9d844cc203fd11f74c55#file-qemu-silicon-mac-virtualization-md)

> Running an intel-based distribution is terrible.

How "terrible"?
As in _"totally frustrating"_ or _"feels like a computer from 2015"_?

**Reply by MstMoonshine** [reply]
A x86 docker container on M1 (docker uses QEMU for containers built with `buildx`) does feel like a computer before 2015.
